samples/retained-instance-store/src/main/java/com/badoo/ribs/samples/retained_instance_store/rib/counter/CounterPresenter.kt

```diff
@@ -0,0 +1,31 @@
+package com.badoo.ribs.samples.retained_instance_store.rib.counter
+
+import androidx.lifecycle.Lifecycle
+import com.badoo.ribs.android.subscribe
+import com.badoo.ribs.core.plugin.ViewAware
+import com.badoo.ribs.samples.retained_instance_store.utils.Clock
+
+interface CounterPresenter
+
+internal class CounterPresenterImpl(private val clock: Clock) : CounterPresenter, ViewAware<CounterView> {
+
+    private var view: CounterView? = null
+
+    override fun onViewCreated(view: CounterView, viewLifecycle: Lifecycle) {
+        viewLifecycle.subscribe(
+                onCreate = {
+                    this@CounterPresenterImpl.view = view
+                    updateCount(clock.elapsedTicks)
+                    clock.bindOnTick(::updateCount)
+                },
+                onDestroy = {
+                    this@CounterPresenterImpl.view = null
+                }
+        )
+        clock.startTimer()
+    }
+
+    private fun updateCount(counter: Int) {
+        view?.updateCount(counter)
```

Comment:
Can we unsubscribe from calling this method in the first place? If the view is null this method shouldn't be called in the first place, rather than not doing something silently.